File: app/src/main/java/com/cyl/musiclake/bean/Playlist.java
package com.cyl.musiclake.bean;

import org.litepal.crud.DataSupport;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * 功能：本地歌单
 * 作者：yonglong on 2016/9/13 21:59
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
public class Playlist extends DataSupport implements Serializable {
    //歌单id
    private String id;
    //歌单名
    private String name;
    //歌单名
    private int count;
    //创建日期
    private long date;
    //描述
    private String des;
    private String order;
    private String coverUrl;

    //歌曲集合
    private List<Music> musicList = new ArrayList<>();

    public Playlist() {
    }

    public Playlist(String id, String name, int count, long date, String order) {
        this.id = id;
        this.name = name;
        this.count = count;
        this.date = date;
        this.order = order;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }

    public long getDate() {
        return date;
    }

    public void setDate(long date) {
        this.date = date;
    }

    public String getDes() {
        return des;
    }

    public void setDes(String des) {
        this.des = des;
    }

    public String getOrder() {
        return order;
    }

    public void setOrder(String order) {
        this.order = order;
    }

    public String getCoverUrl() {
        return coverUrl;
    }

    public void setCoverUrl(String coverUrl) {
        this.coverUrl = coverUrl;
    }

    public List<Music> getMusicList() {
        return musicList;
    }

    public void setMusicList(List<Music> musicList) {
        this.musicList = musicList;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/com/cyl/musiclake/bean/Playlist.kt
package com.cyl.musiclake.bean

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable
import com.cyl.musiclake.common.Constants
import org.litepal.crud.LitePalSupport

@SuppressLint("ParcelCreator")
/**
 * 功能：本地歌单
 * 作者：yonglong on 2016/9/13 21:59
 * 邮箱：643872807@qq.com
 * 版本：2.5
 */
class Playlist() : LitePalSupport(), Parcelable {
    var id: Long = 0
    //歌单id
    var pid: String? = null
    //歌单名
    var name: String? = null
    //歌曲数量
    var total: Long = 0
    //更新日期
    var updateDate: Long = 0
    //更新频率
    var updateFrequency: String? = null
    //创建日期
    var date: Long = 0
    //描述
    var des: String? = null
    //排列顺序
    var order: String? = null
    //封面
    var coverUrl: String? = null
    //类型:本地歌单、在线同步歌单、百度音乐电台、网易云歌单、百度排行榜
    var type: String = Constants.PLAYLIST_LOCAL_ID
    //播放次数
    var playCount: Long = 0
    //歌曲集合
    var musicList = mutableListOf<Music>()

    constructor(parcel: Parcel) : this() {
        id = parcel.readLong()
        pid = parcel.readString()
        name = parcel.readString()
        total = parcel.readLong()
        playCount = parcel.readLong()
        updateDate = parcel.readLong()
        date = parcel.readLong()
        des = parcel.readString()
        order = parcel.readString()
        coverUrl = parcel.readString()
        type = parcel.readString().toString()
    }


    constructor(pid: String?, name: String?) : this() {
        this.pid = pid
        this.name = name
    }

    override fun toString(): String {
        return "Playlist(id=$id, name=$name, total=$total, updateDate=$updateDate, date=$date, des=$des, order=$order, coverUrl=$coverUrl, type=$type, musicList=$musicList)"
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(id)
        parcel.writeString(pid)
        parcel.writeString(name)
        parcel.writeLong(total)
        parcel.writeLong(updateDate)
        parcel.writeLong(date)
        parcel.writeLong(playCount)
        parcel.writeString(des)
        parcel.writeString(order)
        parcel.writeString(coverUrl)
        parcel.writeString(type)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Playlist> {
        override fun createFromParcel(parcel: Parcel): Playlist {
            return Playlist(parcel)
        }

        override fun newArray(size: Int): Array<Playlist?> {
            return arrayOfNulls(size)
        }
    }


}

class MusicToPlaylist : LitePalSupport() {
    var id: Long = 0
    var pid: String? = null
    var mid: String? = null
    var total: Long = 0
    var updateDate: Long = 0
    var createDate: Long = 0
}